functions/Measure-DbaDiskSpaceRequirement.ps1

```diff
@@ -0,0 +1,355 @@
+
+        .PARAMETER Credential
+            Windows credentials to connect via CIM/WMI/PowerShell remoting for MountPoint definition.
+
+        .PARAMETER Consolidate
+            Will summarize space by ComputerName and MountPoints.
+
+        .NOTES
+            Tags: Migration
+
+            Website: https://dbatools.io
+            Copyright: (C) Chrissy LeMaire, clemaire@gmail.com
+            License: GNU GPL v3 https://opensource.org/licenses/GPL-3.0
+
+        .LINK
+            https://dbatools.io/Measure-DbaDiskSpaceRequirement
+
+        .EXAMPLE
+            Measure-DbaDiskSpaceRequirement -SI INSTANCE1 -SD DB1 -DI INSTANCE2 -Consolidate
+
+            Calculate a simple migration with one database with then same name at destination. In this scenario,
+            no space is required since we will recover 448 KB on destination disk.
+
+            DatabaseName1   DatabaseName2   Name1     Name2   SizeKB1 SizeKB2 DiffKB ComputerName MountPoint
+            -------------   -------------   -----     -----   ------- ------- ------ ------------ ----------
+            DB1             DB1             DB1       DB1        3072   -3000     72 INSTANCE2    D:\
+            DB1             DB1             DB1_log   DB1_log     504   -1024   -520 INSTANCE2    D:\
+
+            ComputerName MountPoint RequiredSpaceKB
+            ------------ ---------- ---------------
+            INSTANCE2    D:\                   -448
+
+        .EXAMPLE
+            @([PSCustomObject]@{Source='SQL1';Destination='SQL2';Database='DB1'},
+              [PSCustomObject]@{Source='SQL1';Destination='SQL2';Database='DB2'}
+            ) | Measure-DbaDiskSpaceRequirement -Consolidate
+
+            Using a PSCustomObject with 2 databases to migrate
+
+        .EXAMPLE
+            Import-Csv -Path .\migration.csv -Delimiter "`t" | Measure-DbaDiskSpaceRequirement -Consolidate
+
+            Using a CSV file. You will need a header in migration.csv "Source<tab>Destination<tab>Database"
+
+        .EXAMPLE
+            Invoke-DbaSqlCmd -SqlInstance DBA -Database Migrations -Query 'select Source,Destination,DatabaseName from refresh.Migrations' `
+                | Measure-DbaDiskSpaceRequirement -Consolidate -Verbose
+
+            Using a SQL table. We are DBA after all!
+    #>
+
+    [CmdletBinding()]
+    Param(
+        [Parameter(Mandatory,ValueFromPipelineByPropertyName)]
+        [Alias('SI','SqlInstance')]
+        [string]$Source,
+        [Parameter(Mandatory,ValueFromPipelineByPropertyName)]
+        [Alias('SD','Database','DatabaseName')]
+        [string]$SourceDatabase,
+        [Alias('SC')]
+        [PSCredential]$SourceSqlCredential,
+        [Parameter(Mandatory,ValueFromPipelineByPropertyName)]
+        [Alias('DI')]
+        [string]$Destination, 
+        [Parameter(Mandatory=$false,ValueFromPipelineByPropertyName)]
+        [Alias('DD')]
+        [string]$DestinationDatabase,
```

Comment:
This may need a longer discussion in Slack but I'm not sure I follow the need to have two parameters for the database. 

In my use of this command, it would be to determine if a server has enough space to hold the database from my source server onto the destination server. This is what I get from your synopsis. If that is the case then it does not matter if the destination database already exists or not (in all reality it likely does not). If it did happen to exist I would not care because I could be refreshing a development environment from production, and I'm going overwrite whatever is there.

All that to say in the end you should really only need to have a parameter for `[object[]]$Database` and iterate through each one to see if there is enough space on the destination to hold it.

Reply:
Another use for this command, would be to refresh already existing database in destination environment. Database could be with the same name or not. In both scenario (refresh or new copy), using a database name for destination can be useful as a database with the same name may exist but I really want to create a NEW copy of it.